### PR TITLE
Attempt to fix model editor sidebar

### PR DESCRIPTION
**Pull request by kulyk**
Fixes an issue we encountered on our internal instance. There's a single model for which the metadata editor sidebar doesn't show up.

As far as I know, the only case when it happens is when there's no focused field. [Here's the check](https://github.com/metabase/metabase/blob/1f16ac8374cfc7ba5e701ce56a0a7ba2d4364f26/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx#L118) and [we keep the focused field reference in model editor state](https://github.com/metabase/metabase/blob/1f16ac8374cfc7ba5e701ce56a0a7ba2d4364f26/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx#L239). Usually, it's `undefined` by default and [is set to the first available field once the query completes](https://github.com/metabase/metabase/blob/1f16ac8374cfc7ba5e701ce56a0a7ba2d4364f26/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx#L273). What was odd about the problematic model is that clicking on the column header didn't focus the field either. Turns out, this model's `visualization_settings["table.columns"]` have used `field_ref` instead of expected `fieldRef`. So it made the checks [here](https://github.com/metabase/metabase/blob/1f16ac8374cfc7ba5e701ce56a0a7ba2d4364f26/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx#L216) return nothing, however, we have all the column data.

Fixed by checking for both `fieldRef` and `field_ref` inside the editor 🥴 I'm open to a more global solution for existing questions with mixed naming tho :/ Also fixed what we believe was making some questions end up having incorrectly named `fieldRef` parameters. There's code inside the `Question's` method that is called to sync `visualization_settings["table.columns"]` values with actual columns in query results to ensure they're consistent.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx [hunk 1]
```diff
@@ -210,13 +210,18 @@ function DatasetEditor(props) {
...
-    return orderedColumns
-      .map(
-        col =>
-          columns.find(c => isSameField(c.field_ref, col.fieldRef)) ||
-          virtualCardColumns.find(c => isSameField(c.field_ref, col.fieldRef)),
-      )
-      .filter(Boolean);
+
+    function getFieldFromColumnVizSetting(columnVizSetting) {
+      const fieldRef = columnVizSetting.fieldRef || columnVizSetting.field_ref;
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx:215]: Should we maybe extract this line into a function and put it into `metabase-lib`?

Comment (ranquild) [reply frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx:215]: I think we should also add a comment explaining why this is needed

**Review by ranquild — COMMENTED** [review]
@alxnddr could you take a look? I think if we got corrupted data that there is no other way to fix this, but we need to make sure we take this `field_ref` vs `fieldRef` thing into account everywhere we use column settings

**Comment by kulyk** [comment]
@ranquild @alxnddr from what I can see, I believe the best thing we can do is come up with a Clojure migration that'd look for cards having `table.columns` in viz settings and replace `field_refs` with `fieldRefs`. But I'd love to see @alxnddr's thoughts on this first as I'm not that experienced with visualization settings

**Review by alxnddr — APPROVED** [review]
That's all we can do on FE right now, but I believe we must migrate that on the BE

**Review by ranquild — APPROVED** [review]

**Pull request merged**